eyeshade/workers/reports.js

```diff
@@ -843,6 +991,52 @@ exports.create = create
 exports.publish = publish
 
 exports.workers = {
+/* sent by GET /v1/reports/publisher/{publisher}/referrals
+           GET /v1/reports/publishers/referrals
+
+    { queue            : 'report-publishers-referrals'
+    , message          :
+      { reportId       : '...'
+      , reportURL      : '...'
+      , authority      : '...:...'
+      , format         : 'json' | 'csv'
+      , balance        :  true  | false
+      , summary        :  true  | false
+      , threshold      : probi
+      }
+    }
+ */
+  'report-publishers-referrals':
+    async (debug, runtime, payload) => {
+      const authority = payload.authority
+      const authorized = payload.authorized
+      const balance = payload.balance
+      const format = payload.format
+      const reportId = payload.reportId
+      const summary = payload.summary
+      const thresholdProbi = payload.threshold || 0
+      const verified = payload.verified
+
+      if ((!balance) || (!summary) || (!authorized) || (!verified)) {
+        throw new Error('only summary && balance && authorized && verified is supported')
+      }
+
+      if (format === 'csv') {
```

Comment:
it is probably better to use `if (format !== 'json')`; regardless, i think you may find that being able to get a "csv" formatted file is nice, even if we don't use it as input to the settlement submission tool!